### PR TITLE
Feature/file system restructure

### PR DESCRIPTION
**Pull request by micah-wehrle**
Restructured file system to allow for large divisions in the back end to be split into modules. The primary module is abc-nest.module.ts, which is instantiated in the main.ts file. The AbcNestModule then imports all child modules, such as WeTrackModule and JobsModule. The file structure is as follows:

Screen Shot 2023-04-24 at 5 38 53 PM

Basically everything specific goes into module/[child-module]/ but we can add a universal folder at the same level as the module/ folder.

Sorry if the image still doesn't prove great clarity to the file structure. If you want to sit down and talk it out let me know. Pay attention to the alignment of the file/folder icons. The main.ts file looks like it is within the modules/ folder as there's a gap between the downward arrow and where the main.ts file begins, but you'll note that the icon for main.ts and for the modules/ folder are aligned. This indicates that they are at the same level--that they're both within the same folder.

The code within each module is very barebones just to test that it's working.

### CODE REVIEW

**Comment by micah-wehrle** [comment]
Please bear in mind that the content of the PR isn't as much the focus, as the master branch has almost no functionality, especially not in regards to the front end. This PR is primarily just to reorganize the project to make things easier in the future.

File: src/modules/we-track/controllers/we-track.controller.spec.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import { Test, TestingModule } from '@nestjs/testing';
```
Comment (StreamSG) [inline src/modules/we-track/controllers/we-track.controller.spec.ts:1]: missing a space

Comment (micah-wehrle) [reply src/modules/we-track/controllers/we-track.controller.spec.ts:1]: Oh wow, you'll have to take that up here:
https://github.com/nestjs/nest/issues

Comment (micah-wehrle) [reply src/modules/we-track/controllers/we-track.controller.spec.ts:1]: Actually looks like they all do that

**Comment by micah-wehrle** [comment]
@remindme delete branch around may 10th

**Pull request merged**